### ISSUE TITLE
L2 regularization interacts oddly with lambda layers

### ISSUE THREAD

**Issue by jwlm** [keras-team/keras#12799]
**System information**  
- No custom code  
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  MacOS 10.4.4, also repro'd in Ubuntu 16.4
- TensorFlow backend (yes / no): yes 
- TensorFlow version:  1.13.1
- Keras version:  2.2.4
- Python version:  3.7.3
- CUDA/cuDNN version:  N/A
- GPU model and memory:  N/A

**Describe the current behavior**  

When a trivial intermediate layer is inserted into a larger network with a non-trivial regularizer, the results returned by the compound network change. 

**Describe the expected behavior**  
The intermediate layer subtracts zero from the input, and so can not possibly affect behavior, yet it does.

**Code to reproduce the issue**  
[sorry about the complexity of this case. It's necessary to build all the models and then test them iteratively. -- JWLM]

```
import tensorflow as tf

import keras
from keras.models import Model, Sequential
from keras.layers import Input, Dense, Lambda, Subtract
from keras.regularizers import l2

from sklearn import datasets
from sklearn.preprocessing import OneHotEncoder

from numpy.random import seed, randn

def build_top(digits_train, digits_test, epsilon):
    m = Sequential()
    m.add(Dense(digits_test.shape[1],
                input_shape=(digits_train.shape[1],),
                activation='relu',
                kernel_regularizer=l2(epsilon)))
    m.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['accuracy'])
    
    return m

def build_middle(shape, epsilon):
    m = Sequential()
    m.add(Dense(64, 
                input_shape=(shape,),
                activation='relu',
                kernel_regularizer=l2(epsilon)))
    m.add(Dense(shape))
    
    m.compile(optimizer='adam', loss='cosine_proximity')
    
    return m

def interpolate_network(input_shape, output_shape, layer_model, top_model):
    input_model = Input((input_shape,))
    lm_1 = layer_model(input_model)
    lm_2 = Lambda(lambda x : tf.matmul(x, tf.zeros([input_shape, input_shape])))(lm_1)
    joint_output = Subtract()([input_model, lm_2])
    
    final_model_tensor = top_model(joint_output)
    
    final_model = Model(inputs=input_model, outputs=final_model_tensor)
    final_model.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['accuracy'])
    
    return final_model
    
#def build_digits_net():
    ## import a well-known dataset
digits_dataset = datasets.load_digits()
digits = digits_dataset.data
targets_base = digits_dataset.target
targets = OneHotEncoder(sparse=False, categories='auto').fit_transform(targets_base.reshape(-1, 1))

## Create a random intermediate targert
seed(1)
middle_targets = randn(digits.shape[0], digits.shape[1])

## Build the combined network with no regularization
m_top_0 = build_top(digits, targets, 0.0)
m_top_0.fit(digits, targets)

print('eps=0, top only', m_top_0.evaluate(digits, targets))

m_middle_0 = build_middle(digits.shape[1], 0.0)
m_middle_0.fit(digits, middle_targets)

m_0 = interpolate_network(digits.shape[1], digits.shape[1], m_middle_0, m_top_0)
print('eps=0, combined', m_0.evaluate(digits, targets))
print("--- First two should be equal, and are ---")

## Build the combined network with regularization
m_top_1 = build_top(digits, targets, 1.0)
m_top_1.fit(digits, targets)

print('eps=1, top only', m_top_1.evaluate(digits, targets))

m_middle_1 = build_middle(digits.shape[1], 1.0)
m_middle_1.fit(digits, middle_targets)

m_1 = interpolate_network(digits.shape[1], digits.shape[1], m_middle_1, m_top_1)
print('eps=1, combined', m_1.evaluate(digits, targets))

print("--- Second two should be equal, but aren't---")

print('Environment', 'keras =', keras.__version__, 'tensorflow =', tf.__version__)
```

**Other info / logs**  
Process output:

Epoch 1/1
1797/1797 [==============================] - 1s 329us/step - loss: 4.9922 - acc: 0.0812
1797/1797 [==============================] - 0s 122us/step
eps=0, top only [3.4141229673697673, 0.1613800779242083]
Epoch 1/1
1797/1797 [==============================] - 1s 363us/step - loss: -0.0023
1797/1797 [==============================] - 1s 304us/step
eps=0, combined [3.4141229673697673, 0.1613800779242083]
--- First two should be equal, and are ---
Epoch 1/1
1797/1797 [==============================] - 1s 365us/step - loss: 22.4600 - acc: 0.0801
1797/1797 [==============================] - 0s 137us/step
eps=1, top only [17.648836421913483, 0.08124652198107958]
Epoch 1/1
1797/1797 [==============================] - 1s 397us/step - loss: 44.3188
1797/1797 [==============================] - 1s 329us/step
eps=1, combined [45.07991648755209, 0.08124652198107958]
--- Second two should be equal, but aren't---
Environment keras = 2.2.4 tensorflow = 1.13.1

**Comment by jwlm** [comment]
For the nonce, there's a work-around: if you implement a custom l2 regularizer, then the anomaly disappears.

**Comment by jwlm** [comment]
Sorry, no, that's wrong. The bug persists, even with a custom regularizer.

**Comment by jwlm** [comment]
Also, this behavior persists with a custom regularizer
